cores/esp32/esp32-hal-i2c.c

```diff
@@ -269,86 +283,90 @@ i2c_err_t i2cRead(i2c_t * i2c, uint16_t address, bool addr_10bit, uint8_t * data
     i2cResetCmd(i2c);
 
     //CMD START
-    i2cSetCmd(i2c, 0, I2C_CMD_RSTART, 0, false, false, false);
+    i2cSetCmd(i2c, cmdIdx++, I2C_CMD_RSTART, 0, false, false, false);
 
     //CMD WRITE ADDRESS
```

Comment:
10 bit address are sent high byte, low byte.  Also the address is modified before it is sent.
Philips UM10204 pg.15 3.1.11 10-bit addressing: 
`The first 7 bits of the first byte are the combination of 1111 0xx of which the last two bits (xx) are the two Most-Significant Bits (MSB) of the 10-bit address; the eighth bit of the first byte is the R/W bit that determines the direction of the message.`
So, This section should be changed to:
```c++
// address contains left shifted address with bit0 set for READ
if(addr_10bit){
 i2c->dev->fifo_data.val =((address>>8)& 0x06) | 0xF1; // send a9:a8 and read bit with 1111 0xxR mask
 i2c->dev->fifo_data.val=((address>>1) & 0xFF); // send a7:a0
 }
else{
 i2c->dev->fifo_data.val=(address & 0xFF) ; // send a6:a0 with READ bit
}

Reply:
This code shown is filling the transmission pipeline, it is not about the addressing, but rather the COMMAND buffer that holds the series of I2C actions to perform.

I assume you are referring to the next block for CMD WRITE ADDRESS, I see your point. I didn't even look at this code and just moved it as I was focused on the long reads and my hardware only uses 7 bit addresses. You should raise a separate issue for the addressing issue.

Reply:
@lonerzzz Ok, I'll create a pull with just this 10bit addressing in the _i2cWrite()_ and _i2cRead()_.

Reply:
@stickbreaker, I'll put in your 10 bit fix following mine if @me-no-dev accepts my fix. Let me know if there are concerns.

Reply:
Ya, Just replace the existing code for 10bit addresses in **BOTH** `i2cWrite()` and `i2cRead()` with my snippet.  That way any 10bit address will match Industry specs. 

Chuck.